packages/components/src/navigation/use-navigation-tree-nodes.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+export const useNavigationTreeNodes = () => {
+	const [ nodes, setNodes ] = useState( {} );
+
+	const getNode = ( key ) => nodes[ key ];
+
+	const addNode = ( key, value ) => {
+		const valueCopy = { ...value };
+		delete valueCopy.children;
+
+		setNodes( ( original ) => ( {
+			...original,
+			[ key ]: valueCopy,
+		} ) );
+	};
+
+	const removeNode = ( key ) =>
+		setNodes( ( original ) => {
+			const originalCopy = { ...original };
+			delete originalCopy[ key ];
+			return originalCopy;
+		} );
```

Comment:
Would lodash `omit` work in these functions?
```js
const addNode = ( key, value ) =>
	setNodes( ( original ) => ( {
		...original,
		[ key ]: omit( value, 'children' ),
	} ) );

const removeNode = ( key ) =>
	setNodes( ( original ) => omit( original, key ) );
```

Reply:
Yep. Do you think it'd be readable? `removeNode` looks a bit confusing 🤔

Reply:
I'd personally find it more readable, yeah. 🙂 
We could wait for a second opinion though, it's not a big deal either way.